tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -525,33 +567,55 @@ async fn handle_native_task<
 			};
 			let expires_at = current_block + AUTH_TOKEN_EXPIRATION;
 			let auth_options = AuthOptions { expires_at };
-			let session_claims = AuthTokenClaims::new(
+
+			let access_token_claims = AuthTokenClaims::new(
 				sender.hash().to_string(),
-				AUTH_TOKEN_SESSION_TYPE.to_string(),
+				AUTH_TOKEN_ACCESS_TYPE.to_string(),
 				auth_options.clone(),
 			);
-			let Ok(session_token) = jwt::create(&session_claims, &ctx.jwt_rsa_private_key) else {
+			let Ok(access_token) = jwt::create(&access_token_claims, &ctx.jwt_rsa_private_key)
+			else {
 				let response = NativeTaskResponse::Err(NativeTaskError::AuthTokenCreationFailed);
 				if response_sender.send(response.encode()).is_err() {
 					log::error!("Failed to send response");
 				}
 				return;
 			};
-			let trade_claims = AuthTokenClaims::new(
+			let Ok(user_connect_response) = ctx
+				.pumpx_api
+				.connect_user(&access_token, email.clone(), invite_code, google_code, lang)
+				.await
+			else {
+				log::error!("Failed to connect user");
+				let response = NativeTaskResponse::Err(NativeTaskError::PumpxApiError);
+				if response_sender.send(response.encode()).is_err() {
+					log::error!("Failed to send response");
+				}
+				return;
+			};
+			let id_token_claims = AuthTokenClaims::new(
 				sender.hash().to_string(),
-				AUTH_TOKEN_TRADE_TYPE.to_string(),
+				AUTH_TOKEN_ID_TYPE.to_string(),
 				auth_options,
 			);
-			let Ok(trade_token) = jwt::create(&trade_claims, &ctx.jwt_rsa_private_key) else {
+			let Ok(id_token) = jwt::create(&id_token_claims, &ctx.jwt_rsa_private_key) else {
 				let response = NativeTaskResponse::Err(NativeTaskError::AuthTokenCreationFailed);
 				if response_sender.send(response.encode()).is_err() {
 					log::error!("Failed to send response");
 				}
 				return;
 			};
 
-			let response =
-				NativeTaskResponse::Ok(NativeTaskOk::PumpxJwt { session_token, trade_token });
+			let storage = PumpxAuthTokenIdStorage::new(ctx.storage_db.clone());
+			if storage.insert(sender.hash(), id_token.clone()).is_err() {
+				log::error!("Failed to insert pumpx_auth_token into storage");
```

Comment:
Can we be more specific about logging, like "failed to insert **id** token" - same for other places